lib/parse_config.rb

```diff
@@ -0,0 +1,10 @@
+module FormatParser
+  class ParseConfig < Dry::Struct
```

Comment:
Given that this is _a ton_ of infrastructure just to validate some keyword arguments, do we have a compelling reason to use Dry::Struct? We wanted to use it to make it harder for parsers to return nonsensical information about misparsed files (for example, some image parser returning an image witdh of `negative two million` or some such - which can happen if, for example, you unpack an unsigned int as if it were signed - but not for validating programmer input.

Reply:
I do like this implementation of Dry::Struct for keeping input consistent, but I do agree that if we're going to add a dependency like this we ought to use it to validate output as well.